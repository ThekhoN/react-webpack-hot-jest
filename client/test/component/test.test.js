import React from 'react';
import Test from '../../src/component/test';
import renderer from 'react-test-renderer';

test('Test component is a H2 element with class .test', () => {
  const component = renderer.create(
    <Test/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
