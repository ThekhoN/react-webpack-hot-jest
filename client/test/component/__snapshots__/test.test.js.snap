// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component is a H2 element with class .test 1`] = `
<h2
  className="test normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test Component Updated HMR with Jest
</h2>
`;

exports[`Test component is a H2 element with class .test 2`] = `
<h2
  className="test hovered"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test Component Updated HMR with Jest
</h2>
`;

exports[`Test component is a H2 element with class .test 3`] = `
<h2
  className="test normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Test Component Updated HMR with Jest
</h2>
`;
